backup/src/main/scala/backup/Lambda.scala

```diff
@@ -5,41 +5,47 @@ import akka.stream.ActorMaterializer
 import backup.logging.BackupLogging
 import com.amazonaws.auth.profile.ProfileCredentialsProvider
 import com.amazonaws.auth.{AWSCredentialsProviderChain, InstanceProfileCredentialsProvider}
+import play.api.libs.ws.WSClient
 import play.api.libs.ws.ahc.AhcWSClient
 
-import scala.concurrent.ExecutionContext
-import scala.util.{Failure, Success}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
 
 object Lambda extends BackupLogging {
 
   implicit val executionContext: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
-  implicit val actorSystem = ActorSystem()
-  implicit val actorMaterializer = ActorMaterializer()
 
-  val credentials = new AWSCredentialsProviderChain(
-    new ProfileCredentialsProvider("mobile"),
-    InstanceProfileCredentialsProvider.getInstance
-  )
+  private def withWsClient[A](block: WSClient => Future[A]): Future[A] = {
+    implicit val actorSystem: ActorSystem = ActorSystem()
+    implicit val actorMaterializer: ActorMaterializer = ActorMaterializer()
+    val ws = AhcWSClient()
 
-  val config = Configuration.load(credentials)
+    block(ws)
+      .andThen { case _ => ws.close() }
+      .andThen { case _ => actorSystem.terminate() }
+  }
```

Comment:
Nice. Does this fix the backup error thing or is that a separate issue?

Reply:
It makes running the lambda locally nicer as if an exception happens the whole jvm blocks and waits. I've also set `fork := true` just to be extra sure I don't have to close my sbt session